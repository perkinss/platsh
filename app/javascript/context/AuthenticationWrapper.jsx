import React, { useEffect, useState, useContext } from 'react';
import User from '../model/user'

export const RootAuthContext = React.createContext();
export default ({ children }) => {
  let anyAuthVal,anyAuthData;
  try {
    anyAuthVal=JSON.parse(window.sessionStorage.getItem('authenticated'));
  }
  catch(err) {
    anyAuthVal=false;
  }
  try {
    anyAuthData = JSON.parse(window.sessionStorage.getItem('authData'))
  }
  catch(err) {
    anyAuthData = null;
  }
  const guestUser = new User({})
  const prevAuth = (anyAuthVal ? anyAuthVal : false);
  const prevAuthData = (anyAuthData ? anyAuthData : null);
  const prevUser = (anyAuthData ? new User(anyAuthData) : guestUser);
  const [authenticated, setAuthenticated] = useState(prevAuth);
  const [authData, setAuthData] = useState(prevAuthData);
  const [currentUser, setCurrentUser] = useState(prevUser);
  useEffect(
    () => {
      window.sessionStorage.setItem('authenticated', JSON.stringify(authenticated));
      window.sessionStorage.setItem('authData', (authData ? JSON.stringify(authData):null));
      setCurrentUser(authData ? new User(authData) : guestUser)
    },
    [authenticated, authData]
  );
  const defaultContext = {
    authenticated, setAuthenticated,
    authData, setAuthData,
    currentUser, setCurrentUser
  };
  return (
    <RootAuthContext.Provider value={defaultContext}>
      {children}
    </RootAuthContext.Provider>
  );
};
