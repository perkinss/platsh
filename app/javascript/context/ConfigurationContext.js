import React from 'react'

export const configuration = {
    sections: [
        {
            id: null,
            courses: [],
            name: '',
            assessment: [
                {
                    type: 'test',
                    name: '',
                    tasks: [
                    {
                        name: '',
                        available_standards: [
                            {
                                id: -1,
                                level: ''
                            }
                        ],
                        competencies: [

                        ]
                    },
                    ]
                }
            ],
            enrollees: [
                1, 2, 3, 22
            ]
        },
    ],

}


const ConfigurationContext =  React.createContext(configuration);

export { ConfigurationContext }
