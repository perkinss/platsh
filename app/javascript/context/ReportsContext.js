import React from 'react'

export const reports = {
    students: [],
    sections: [],
    selected_section: null,
    selected_student: null,
    section_overview_marks: {},
    section_competency_scores: {},
    student_section_marks: {},
    heat_map_colors: []
}


const ReportsContext =  React.createContext(reports);

export { ReportsContext }
