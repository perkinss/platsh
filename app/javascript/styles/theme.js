import {createMuiTheme} from "@material-ui/core/styles";

const theme = createMuiTheme({
    palette: {
        primary: {
            main: '#512da8',
        },
        secondary: {
            main: '#ab47bc',
        },
        type: 'dark'
    },
    slider: {
        selectionColor: '#F76B1C',
        handleFillColor: '#F76B1C',
        thumb: 'secondary'
    }

})

export default theme
