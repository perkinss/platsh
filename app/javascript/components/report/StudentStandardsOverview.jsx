import React, {useContext} from 'react'
import {Typography, CircularProgress} from '@material-ui/core'
import StudentContentBarChart from "./StudentContentBarChart";
import {ReportsContext} from "../../context/ReportsContext";
import {isNullOrUndefined} from "../../helpers/object_helper";
import {useTheme} from "@material-ui/styles";

export default function StudentStandardsOverview(props) {
    const { student, section, tabIndex, loading, standardReport = {}, observedStandards = {} } = props

    const { state } = useContext(ReportsContext)
    const theme = useTheme()

    let courseId = section.courses[tabIndex]?.id
    let series, labels, data

    const noMarkScore = (topicId) => {
        const course_mark = standardReport.course_marks && standardReport.course_marks.find(course_detail => course_detail.id === topicId)
        return course_mark && course_mark.marks.every( mark => !(mark.id in observedStandards) ) ? null : 0
    }

    if (state && state.topic_averages && state.topic_averages[section.id] && state.topic_averages[section.id][student.id]) {
        data = state.topic_averages[section.id][student.id].find(course => course.id === courseId)
        if (data) {
            series = [{ name: "Topic total", data: data.course_marks.map ( content => content.mark || noMarkScore(content.id) ) } ]
            labels = data.course_marks.map ( content => content.name )
        }
    }

    if (loading || isNullOrUndefined(data)) {
        return (
            <div style={{padding: theme.spacing(5,0,2,0)}}>
                <Typography variant={'h5'}>Course Topics Overview</Typography>
                <CircularProgress color={'primary'} />
            </div>
        )
    } else {
        return (
            <div style={{padding: theme.spacing(5,0,8,0)}}>
                <Typography variant={'h5'}>Course Topics Overview</Typography>
                <StudentContentBarChart series={series} categories={labels}/>
            </div>
        )
    }
}