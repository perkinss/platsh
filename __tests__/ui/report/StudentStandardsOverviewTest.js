import React from 'react'
import { createRender } from '@material-ui/core/test-utils';
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import expect from 'expect'
import theme from "../../../app/javascript/styles/theme";
import { ThemeProvider } from '@material-ui/styles'
import StudentStandardsOverview from "../../../app/javascript/components/report/StudentStandardsOverview";
import { ReportsContext } from "../../../app/javascript/context/ReportsContext";

Enzyme.configure({ adapter: new Adapter() });

describe('<StudentStandardsOverview />', () => {
    let render;
    let classes = {
        content: {}
    }
    let state = {
        topic_averages: {
            3: {
                5:
                    [{ id: 25,
                        course_marks: [
                            {
                                id: 111,
                                name: "Applications of Integration",
                                mark: null,
                                weight: 5
                            }
                        ]
                    }]
            }
        }
    }
    let section = {
        id: 3,
        courses: {
            1: {
                id: 25
            }
        }
    }
    let student = {
        id: 5, name: "Ed Shearan", unique_id: "edshe@ran.com", created_at: "2019-07-07T05:07:00.505Z", updated_at: "2019-07-07T05:07:00.505Z"
    }
    beforeAll(() => {
        render = createRender();
    })

    it('should match the snapshot', () => {
        const wrapper = render(
            <ThemeProvider theme={theme}>
                <ReportsContext.Provider value={{dispatch: () => {}, state: {} }} >
                    <StudentStandardsOverview classes={classes} student={student} loading={true} tabIndex={1} section={section} />
                </ReportsContext.Provider>
            </ThemeProvider>
                );
        expect(wrapper).not.toBeNull()
        expect(wrapper).toMatchSnapshot()
        expect(wrapper.html()).toContain("CircularProgress")
        expect(wrapper.find('#student-content-chart').html()).toBeNull()
    });

    it('should have the student details on first render', () => {
        const wrapper = render(
            <ThemeProvider theme={theme}>
                <ReportsContext.Provider value={{dispatch: () => {}, state: state }} >
                    <StudentStandardsOverview classes={classes} student={student} section={section} tabIndex={1} />
                </ReportsContext.Provider>
            </ThemeProvider>
        );
        let result = wrapper.html()
        expect(result).toContain("Topics Overview")
        let chart = wrapper.find("#student-content-chart")
        expect(chart.html()).not.toBeNull()
    });

})