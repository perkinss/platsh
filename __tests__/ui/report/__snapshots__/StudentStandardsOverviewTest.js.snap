// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<StudentStandardsOverview /> should match the snapshot 1`] = `
initialize {
  "0": Node {
    "attribs": Object {
      "style": "padding:40px 0px 16px 0px",
    },
    "children": Array [
      Node {
        "attribs": Object {
          "class": "MuiTypography-root MuiTypography-h5",
        },
        "children": Array [
          Node {
            "data": "Course Topics Overview",
            "next": null,
            "parent": [Circular],
            "prev": null,
            "type": "text",
          },
        ],
        "name": "h5",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": Node {
          "attribs": Object {
            "class": "MuiCircularProgress-root MuiCircularProgress-colorPrimary MuiCircularProgress-indeterminate",
            "role": "progressbar",
            "style": "width:40px;height:40px",
          },
          "children": Array [
            Node {
              "attribs": Object {
                "class": "MuiCircularProgress-svg",
                "viewBox": "22 22 44 44",
              },
              "children": Array [
                Node {
                  "attribs": Object {
                    "class": "MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate",
                    "cx": "44",
                    "cy": "44",
                    "fill": "none",
                    "r": "20.2",
                    "stroke-width": "3.6",
                  },
                  "children": Array [],
                  "name": "circle",
                  "namespace": "http://www.w3.org/2000/svg",
                  "next": null,
                  "parent": [Circular],
                  "prev": null,
                  "type": "tag",
                  "x-attribsNamespace": Object {
                    "class": undefined,
                    "cx": undefined,
                    "cy": undefined,
                    "fill": undefined,
                    "r": undefined,
                    "stroke-width": undefined,
                  },
                  "x-attribsPrefix": Object {
                    "class": undefined,
                    "cx": undefined,
                    "cy": undefined,
                    "fill": undefined,
                    "r": undefined,
                    "stroke-width": undefined,
                  },
                },
              ],
              "name": "svg",
              "namespace": "http://www.w3.org/2000/svg",
              "next": null,
              "parent": [Circular],
              "prev": null,
              "type": "tag",
              "x-attribsNamespace": Object {
                "class": undefined,
                "viewBox": undefined,
              },
              "x-attribsPrefix": Object {
                "class": undefined,
                "viewBox": undefined,
              },
            },
          ],
          "name": "div",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": null,
          "parent": [Circular],
          "prev": [Circular],
          "type": "tag",
          "x-attribsNamespace": Object {
            "class": undefined,
            "role": undefined,
            "style": undefined,
          },
          "x-attribsPrefix": Object {
            "class": undefined,
            "role": undefined,
            "style": undefined,
          },
        },
        "parent": [Circular],
        "prev": null,
        "type": "tag",
        "x-attribsNamespace": Object {
          "class": undefined,
        },
        "x-attribsPrefix": Object {
          "class": undefined,
        },
      },
      Node {
        "attribs": Object {
          "class": "MuiCircularProgress-root MuiCircularProgress-colorPrimary MuiCircularProgress-indeterminate",
          "role": "progressbar",
          "style": "width:40px;height:40px",
        },
        "children": Array [
          Node {
            "attribs": Object {
              "class": "MuiCircularProgress-svg",
              "viewBox": "22 22 44 44",
            },
            "children": Array [
              Node {
                "attribs": Object {
                  "class": "MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate",
                  "cx": "44",
                  "cy": "44",
                  "fill": "none",
                  "r": "20.2",
                  "stroke-width": "3.6",
                },
                "children": Array [],
                "name": "circle",
                "namespace": "http://www.w3.org/2000/svg",
                "next": null,
                "parent": [Circular],
                "prev": null,
                "type": "tag",
                "x-attribsNamespace": Object {
                  "class": undefined,
                  "cx": undefined,
                  "cy": undefined,
                  "fill": undefined,
                  "r": undefined,
                  "stroke-width": undefined,
                },
                "x-attribsPrefix": Object {
                  "class": undefined,
                  "cx": undefined,
                  "cy": undefined,
                  "fill": undefined,
                  "r": undefined,
                  "stroke-width": undefined,
                },
              },
            ],
            "name": "svg",
            "namespace": "http://www.w3.org/2000/svg",
            "next": null,
            "parent": [Circular],
            "prev": null,
            "type": "tag",
            "x-attribsNamespace": Object {
              "class": undefined,
              "viewBox": undefined,
            },
            "x-attribsPrefix": Object {
              "class": undefined,
              "viewBox": undefined,
            },
          },
        ],
        "name": "div",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": null,
        "parent": [Circular],
        "prev": Node {
          "attribs": Object {
            "class": "MuiTypography-root MuiTypography-h5",
          },
          "children": Array [
            Node {
              "data": "Course Topics Overview",
              "next": null,
              "parent": [Circular],
              "prev": null,
              "type": "text",
            },
          ],
          "name": "h5",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": [Circular],
          "parent": [Circular],
          "prev": null,
          "type": "tag",
          "x-attribsNamespace": Object {
            "class": undefined,
          },
          "x-attribsPrefix": Object {
            "class": undefined,
          },
        },
        "type": "tag",
        "x-attribsNamespace": Object {
          "class": undefined,
          "role": undefined,
          "style": undefined,
        },
        "x-attribsPrefix": Object {
          "class": undefined,
          "role": undefined,
          "style": undefined,
        },
      },
    ],
    "name": "div",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": Node {
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
    },
    "prev": null,
    "type": "tag",
    "x-attribsNamespace": Object {
      "style": undefined,
    },
    "x-attribsPrefix": Object {
      "style": undefined,
    },
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "xml": false,
  },
}
`;
